src/mission_start.cpp

```diff
@@ -1941,3 +1948,145 @@ void mission_start::reveal_lab_train_depot( mission *miss )
 }
+
+void mission_start_t::set_reveal( const std::string &terrain )
+{
+    const auto start_func = [ terrain ]( mission * miss ) {
+        reveal_target( miss, terrain );
+    };
+    start_funcs.push_back( start_func );
+}
+
+void mission_start_t::set_reveal_any( JsonArray &ja )
+{
+    std::vector<std::string> terrains;
+    while( ja.has_more() ) {
+        std::string terrain = ja.next_string();
+        terrains.push_back( terrain );
+    }
+    const auto start_func = [ terrains ]( mission * miss ) {
+        reveal_any_target( miss, terrains );
+    };
+    start_funcs.push_back( start_func );
+}
+
+
+void mission_start_t::set_target_om( JsonObject &jo, bool random = false )
+{
+    int reveal_rad = 1;
+    bool must_see = false;
+    int target_z = 0;
+    int range = 5;
+    if( !jo.has_string( "om_ter" ) ) {
+        return;
+    }
+    std::string omter = jo.get_string( "om_ter" );
+    if( jo.has_int( "reveal_rad" ) ) {
+        reveal_rad = std::max( 1, jo.get_int( "reveal_rad" ) );
+    }
+    if( jo.has_bool( "must_see" ) ) {
+        must_see = jo.get_bool( "must_see" );
+    }
+    if( jo.has_int( "zlevel" ) ) {
+        target_z = jo.get_int( "zlevel" );
+    }
+    if( jo.has_int( "range" ) ) {
+        range = jo.get_int( "range" );
+    }
+    if( random ) {
+        const auto start_func = [ omter, reveal_rad, must_see, range ]( mission * miss ) {
+            target_om_ter_random( omter, reveal_rad, miss, must_see, range );
+        };
+        start_funcs.push_back( start_func );
+    } else {
+        const auto start_func = [ omter, reveal_rad, must_see, target_z ]( mission * miss ) {
+            target_om_ter( omter, reveal_rad, miss, must_see, target_z );
+        };
+        start_funcs.push_back( start_func );
+    }
+}
+
+void mission_start_t::set_target_om_or_create( JsonObject &jo )
+{
+    int reveal_rad = 1;
+    bool must_see = false;
+    int range = 5;
+    if( !( jo.has_string( "om_ter" ) && jo.has_string( "om_spec" ) &&
+           jo.has_string( "om_replace_ter" ) ) ) {
+        return;
+    }
+    std::string omter = jo.get_string( "om_ter" );
+    std::string om_spec = jo.get_string( "om_spec" );
+    std::string om_replace_ter = jo.get_string( "om_replace_ter" );
+    if( jo.has_int( "reveal_rad" ) ) {
+        reveal_rad = std::max( 1, jo.get_int( "reveal_rad" ) );
+    }
+    if( jo.has_bool( "must_see" ) ) {
+        must_see = jo.get_bool( "must_see" );
+    }
+    if( jo.has_int( "range" ) ) {
+        range = jo.get_int( "range" );
+    }
+    const auto start_func = [ omter, reveal_rad, must_see, range, om_spec,
+           om_replace_ter ]( mission * miss ) {
+        target_om_ter_random_or_create( omter, reveal_rad, miss, must_see, range,
+                                        om_spec, om_replace_ter );
+    };
+    start_funcs.push_back( start_func );
+}
+
+void mission_start_t::load( JsonObject &jo )
+{
+    if( jo.has_string( "reveal_om_ter" ) ) {
```

Comment:
Wouldn't it be better to support only one node `reveal_om_ter` and check if there is a string or array inside of json node in the code?

Reply:
Good idea, fixed.